cmd/controller/main.go

```diff
@@ -44,14 +46,25 @@ const (
 )
 
 func main() {
+	var logLevel string
+
 	flag.IntVar(&ctrlConcurrency, "concurrency", 10, "Max concurrent reconciles")
 	flag.StringVar(&ctrlNamespace, "namespace", "", "Namespace to watch")
 	flag.BoolVar(&allowSharedServiceAccount, "dangerous-allow-shared-service-account",
 		false, "If set to true, allow use of shared service account instead of per-app service accounts")
 	flag.BoolVar(&enablePprof, "dangerous-enable-pprof", false, "If set to true, enable pprof on "+pprofListenAddr)
+	flag.StringVar(&logLevel, "loglevel", zapcore.InfoLevel.String(), "Log level: debug/info/warn/error/dpanic/panic/fatal")
 	flag.Parse()
 
-	logf.SetLogger(zap.Logger(false))
+	level := uberzap.NewAtomicLevel()
+	if err := level.UnmarshalText([]byte(logLevel)); err != nil {
+		level.SetLevel(zapcore.InfoLevel)
```

Comment:
Would we want to log something about this being unsuccessful?

Reply:
I have added a log entry in case the unmarshaling of the log level fails.